ci/taos/webapp/broken_arrow_proceed.php

```diff
@@ -0,0 +1,57 @@
+<?php
+/**
+ * Copyright (c) 2018 Samsung Electronics Co., Ltd. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ *    @file   broken_arrow_proceed.php
+ *    @brief  A runner to do "broken arrow" command.
+ *
+ */
+
+// get input date from FORM tag.
+$id = $_POST['id'];
+$pass = $_POST['pass'];
+
+// Check if id and password is typed.
+if ($id == "" || $pass == "")
+   die ("Unable to do the broken arrow. Please type a ID and Password.")
+
+// Read a JSON file.
+$string = file_get_contents("../config/config-webhook.json");
+$json_config = json_decode($string);
+
+// Get id and passowrd from json file.
+$ba_id = $json_config->broken_arrow->id;
+$ba_pass = $json_config->broken_arrow->pass;
+
+// Run the "BROKEN ARROW" command.
+if ( $id == $ba_id && $pass == $ba_pass) {
+    //$cmd = "cat $password | su -c 'shutdown -r now'";
+    $cmd = "sudo shutdown -r now";
```

Comment:
what is different with 'sudo reboot -f'?

Reply:
difference is a security. As you know, we have to do not use the password of system account directly.